File: src/lessons/welcome/MethodsPictureEntity.java
package lessons.welcome;

import java.awt.Color;

/* The suppress warning is sometimes mandatory for student code to compile cleanly */
@SuppressWarnings("unused")
public class MethodsPictureEntity extends jlm.universe.bugglequest.SimpleBuggle {

	/* BEGIN SOLUTION */
	void mark() {
		brushDown();
		brushUp();
	}
		
	void makeV(Color c) {
		setBrushColor(c);
		forward();
		mark();
		
		forward();
		turnLeft();
		forward();
		mark();
		
		backward();
		turnRight();
		forward();
		mark();
		
		forward();
		turnLeft();
	}

	
	
	public void run() {
		makeV(Color.yellow);
		makeV(Color.red);
		makeV(Color.blue);
		makeV(Color.green);
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/MethodsPictureEntity.py
import java.awt.Color as Color

# BEGIN SOLUTION
def mark():
   brushDown()
   brushUp()

def makeV(c):
   setBrushColor(c)
   forward()
   mark()
   forward()
   turnLeft()
   forward()
   mark()
   backward()
   turnRight()
   forward()
   mark()   
   forward()
   turnLeft()


makeV(Color.yellow)
makeV(Color.red)
makeV(Color.blue)
makeV(Color.green)
# END TEMPLATE
